### PR TITLE
MNT: fix up from mds changes

### PR DESCRIPTION
**Pull request by tacaswell**
- Deal with new header documents
 - deal with docs being read-only

### CODE REVIEW

File: suitcase.py [hunk 1]
```diff
@@ -20,44 +17,49 @@ def export(headers, filename):
...
+
+                desc_group = group.create_group(descriptor['uid'])
+
                 data_keys = descriptor.pop('data_keys')
                 _safe_attrs_assignment(desc_group, descriptor)
+
                 events = list(find_events(descriptor=descriptor))
                 event_times = [e['time'] for e in events]
-                desc_group.create_dataset('event_times', event_times)
+                desc_group.create_dataset('time', data=event_times)
```
Comment (ericdill) [inline suitcase.py]: I do not particularly care whether this is named `event_times` or `time`, but I am curious why this was renamed.

File: suitcase.py [hunk 2]
```diff
@@ -20,44 +17,49 @@ def export(headers, filename):
...
+            header = dict(header)
             try:
-                descriptors = header.pop('event_descriptors')
+                descriptors = header.pop('descriptors')
             except KeyError:
                 warnings.warn("Header with uid {header.uid} contains no "
                               "data.".format(header), UserWarning)
                 continue
-            top_group_name = repr(header).replace(' ', '_')[1:-1]
+            top_group_name = header['start']['uid']
```
Comment (ericdill) [inline suitcase.py]: I like this change. Simply being the uid of the run start feels cleaner

**Comment by ericdill** [comment]
For my own knowledge, when would `node.attrs[key] = value` fail?

**Comment by ericdill** [comment]
Looks good. Thanks for doing this

File: suitcase.py (hunk 1 above)
Comment (tacaswell) [inline suitcase.py]: To match the name in the event documents so we don't have to keep track of any mappings.

**Comment by tacaswell** [comment]
Most of h5py works by casting things to numpy and then mapping from numpy -> hdf5's type system.  Anything that fails along that path can't be shove in without custom serialization logic.  The simplest is a dict:  

```python
In [7]: import h5py

In [8]: f = h5py.File(':memory:')

In [9]: f.attrs
Out[9]: <Attributes of HDF5 object at 140225191548680>

In [10]: f.attrs['foo'] = {'a': 1}
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-10-2a6fb928d0e9> in <module>()
----> 1 f.attrs['foo'] = {'a': 1}

h5py/_objects.pyx in h5py._objects.with_phil.wrapper (-------src-dir-------/h5py/_objects.c:2458)()

h5py/_objects.pyx in h5py._objects.with_phil.wrapper (-------src-dir-------/h5py/_objects.c:2415)()

/home/tcaswell/.virtualenvs/dd_py3k/lib/python3.4/site-packages/h5py/_hl/attrs.py in __setitem__(self, name, value)
     85         use the methods create() and modify().
     86         """
---> 87         self.create(name, data=value, dtype=base.guess_dtype(value))
     88 
     89     @with_phil

/home/tcaswell/.virtualenvs/dd_py3k/lib/python3.4/site-packages/h5py/_hl/attrs.py in create(self, name, data, shape, dtype)
    161             # Make HDF5 datatype and dataspace for the H5A calls
    162             if use_htype is None:
--> 163                 htype = h5t.py_create(original_dtype, logical=True)
    164                 htype2 = h5t.py_create(original_dtype)  # Must be bit-for-bit representation rather than logical
    165             else:

h5py/h5t.pyx in h5py.h5t.py_create (-------src-dir-------/h5py/h5t.c:15483)()

h5py/h5t.pyx in h5py.h5t.py_create (-------src-dir-------/h5py/h5t.c:15314)()

h5py/h5t.pyx in h5py.h5t.py_create (-------src-dir-------/h5py/h5t.c:15232)()

TypeError: Object dtype dtype('O') has no native HDF5 equivalent
```

but custom objects and unicode strings also fail.

**Comment by tacaswell** [comment]
I also view this as a MVP, others are more than welcome to write dialects that output the data formatted how ever they want.

**Comment by ericdill** [comment]
Ah cool, thanks for the clear example

File: suitcase.py (hunk 1 above)
Comment (ericdill) [inline suitcase.py]: Makes sense. :+1:

**Pull request merged**